src/client/ua_client.c

```diff
@@ -20,6 +21,15 @@
 static void UA_Client_init(UA_Client* client, UA_ClientConfig config) {
     memset(client, 0, sizeof(UA_Client));
     client->channel.connection = &client->connection;
+    client->channel.endpoint = (UA_Endpoint*)UA_malloc(sizeof(UA_Endpoint));
```

Comment:
Hmm, well if reset nulls the pointer, then we need to allocate a new one in init, no?
Or is it guaranteed, that new is called before anything else? Then we could move this to new and delete.